src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/BodyModelBinder.cs

```diff
@@ -132,7 +148,21 @@ public BodyModelBinder(IList<IInputFormatter> formatters, IHttpRequestStreamRead
                     return;
                 }
 
-                bindingContext.Result = ModelBindingResult.Success(model);
+                if (_isBindingRequired && model == null)
```

Comment:
I'm with @dougbu on this - maybe if we were starting from scratch there'd be a case for not having 108, but it seems like an unnecessary breaking change to remove it.

Removing it means that everyone's custom `InputFormatter` subclasses would start getting calls to `ReadRequestBodyAsync` in cases where they wouldn't previously (when the `ContentLength` is zero), and everyone would have to duplicate the same logic about checking whether empty input is configured as allowed and if so, producing a default value for the result.

Given that (1) we have back-compatibility as a (soft) goal, and (2) the existing base-class behaviour is correct in virtually all cases, and (3) developers can always override `ReadAsync` completely if they really want to change this, I don't see how it's a good idea to push this check downstream.

> Put another way, the individual input formatters will be responsible for deciding whether or not they received data

Yes, that, *plus* they are responsible for providing a default value for the model in cases where there's no input yet empty input is allowed. That's the existing 1.x design.